kratos/utilities/binbased_fast_point_locator.h

```diff
@@ -109,11 +107,40 @@ class BinBasedFastPointLocator
     explicit BinBasedFastPointLocator(ModelPart& rModelPart)
         : mrModelPart(rModelPart)
     {
+        // NOTE: Should call UpdateSearchDatabase() to initialize the search database
+        // UpdateSearchDatabase();
+    }
+
+    /**
+     * @brief This is the default constructor (with cell size)
+     * @param rModelPart The model part of the mesh used in the search
+     * @param CellSize The current size of the cell used for search
+     */
+    explicit BinBasedFastPointLocator(
+        ModelPart& rModelPart,
+        const double CellSize
+        )
+        : mrModelPart(rModelPart),
+          mCellSize(CellSize)
+    {
+        UpdateSearchDatabaseAssignedSize(mCellSize);
     }
 
     /// Destructor.
     virtual ~BinBasedFastPointLocator() = default;
 
+    /// Copy constructor.
+    BinBasedFastPointLocator(BinBasedFastPointLocator const& rOther)
+        : mrModelPart(rOther.mrModelPart),
+          mCellSize(rOther.mCellSize)
```

Comment:
Why not calling BinsDynamicObject's: `rOther.GetCellSize()` instead of forwarding that parameter to this class?

Reply:
Huuuum

Reply:
mCellSize in BinsObjectDynamic is an array, not a double, not possible

Reply:
but I don't get what you are trying to do, mCellSize is an array because the cells of the Bins are cubes.

If you are copying the underlying Bins from the utility, and you only forward the int, then you are imposing the cells to have the same size, which is not correct in many cases (elongated models, planes in 3D, etc...),.

By passing the double, you are wrongly initializing the bins using this constructor:

```c++
  BinsObjectDynamic (const PointType& MinPoint, const PointType& MaxPoint, SizeType NumPoints)
      : mObjectsSize(0), mObjectsBegin(0), mObjectsEnd(0) {

    for(SizeType i = 0; i < Dimension; i++) {
      mMinPoint[i] = MinPoint[i];
      mMaxPoint[i] = MaxPoint[i];
    }

    CalculateCellSize(NumPoints);                                 // Calculate number of Cells
    AllocateContainer();                                         // Allocate cell list
  }
```

Which allocates the number of points, not the one that allocates the size of the cells.

Reply:
I was simply trying to do a consistent copy constructor with the existent methods